tensorflow/compiler/xla/service/gpu/cudnn_conv_rewriter.cc

```diff
@@ -248,7 +246,29 @@ std::tuple<bool, Window, ConvolutionDimensionNumbers> MatchBackwardFilter(
     backward_conv_dnums.add_kernel_spatial_dimensions(output_spatial_dims[i]);
   }
 
-  return std::make_tuple(true, backward_conv_window, backward_conv_dnums);
+  HloInstruction* lhs = conv->mutable_operand(0);
+  if (conv->feature_group_count() == 1) {
+    return std::make_tuple(true, backward_conv_window, backward_conv_dnums,
+                           lhs);
+  }
+  Shape new_shape = lhs->shape();
+
+  int64 input_batch_dimension = backward_conv_dnums.input_batch_dimension();
+  int64 input_feature_dimension = backward_conv_dnums.input_feature_dimension();
+
+  int64 input_batch = new_shape.dimensions(input_batch_dimension);
+  int64 input_feature = new_shape.dimensions(input_feature_dimension);
+
+  // Ensure that input_batch is exact multiple of conv->feature_group_count()
+  CHECK_EQ(input_batch % conv->feature_group_count(), 0);
```

Comment:
Please output a message briefly explaining this invariant should the CHECK fail.
CHECK_EQ(...) << "..."

Reply:
Done